src/components/Session/Chat.vue

```diff
@@ -1,79 +1,112 @@
-						<div class="avatar" v-bind:style="message.avatarStyle"></div>
-						<div class="contents">
-              <div class="name">
-								{{message.name}}
-							</div>
-							{{message.contents}}
-              <div class="time">
-								{{message.time}}
-							</div>
+<div class="chat">
+  
+  <div class="header">Chat</div>
+
+  <div class="message-box">
+
+    <div class="chat-warning">
+      Messages cannot contain personal information
+      <span class="chat-warning__close" @click="hideModerationWarning">×</span>
+    </div>
+
+		<div class="messages">
+			<template v-for="message in messages">
+
+				<div class="message" :class="message.name === user.firstname ? 'left' : 'right'">
+					<div class="avatar" :style="message.avatarStyle"></div>
+					<div class="contents">
+            <div class="name">
+							{{message.name}}
+						</div>
+						{{message.contents}}
+            <div class="time">
+							{{message.time}}
 						</div>
 					</div>
-				</template>
-			</div>
-    </div>
-    <textarea v-on:keyup.enter="sendMessage" v-model="newMessage" placeholder="Type here."></textarea>
+				</div>
+
+			</template>
+		</div>
   </div>
+
+  <textarea @keyup.enter="sendMessage" v-model="newMessage" placeholder="Type here..."></textarea>
+
+</div>
 </template>
 
-<script>
 
-import $ from 'jquery'
-import moment from 'moment'
+<script>
+import moment from 'moment';
 
-import UserService from 'src/services/UserService'
-import SessionService from 'src/services/SessionService'
+import UserService from 'src/services/UserService';
+import SessionService from 'src/services/SessionService';
+import ModerationService from 'src/services/ModerationService';
 
 const DEFAULT_AVATAR_URL = 'static/defaultAvatar@2x.png';
 
 export default {
-	data(){
-    var user = UserService.getUser();
+	data() {
 		return {
-      user: user,
+      user: UserService.getUser(),
       messages: [],
 			currentSession: SessionService.currentSession,
 			newMessage: ''
 		}
 	},
+
   methods: {
-    sendMessage: function() {
-      var message = this.newMessage;
+    showModerationWarning() {
+      document.querySelector('.chat-warning').style.top = 0;
+    },
+    hideModerationWarning() {
+      document.querySelector('.chat-warning').style.top = '';
+    },
 
+    showNewMessage(message) {
       this.$socket.emit('message', {
-				sessionId: this.currentSession.sessionId,
+        sessionId: this.currentSession.sessionId,
         user: UserService.getUser(),
         message: message
       });
-
       this.newMessage = '';
     },
-    leftRightMessage: function(message) {
-      if (message.name == this.user.firstname) {
-        return 'left';
-      } else {
-        return 'right';
+
+    sendMessage() {
+      let message = this.newMessage.slice(0,-1);
+
+      if (message != '') {
+
+        ModerationService.checkIfMessageIsClean(this, message).then((isClean) => {
+
+          if (isClean != null) {
```

Comment:
Logic is:

` if (isClean === true) { this.showNewMessage(message) } else {this.showModerationWarning()}`

Also use strict equality

Reply:
@stackptr This was also what I thought in the beginning, but then I realized that this logic doesn't work because if for some reason the cleanliness of the message can't be determined, then `isClean` is going to be `null` (see [this line](https://github.com/UPchieve/web/pull/72/files#diff-4fbc7cb1c3b6d337eeaf0fd91c9284fbR6)), and then the user won't be able to send any message (since `isClean` will keep being `null`, ergo different than `true`), which is undesirable because it breaks the UX completely.

A valid option to avoid using two conditionals would be:
`if (isClean === false) { this.showModerationWarning(); } else { this.showNewMessage(message); }`

Would you rather I use this option or leave it as it is?

Reply:
Since we control the `ModerationService` method and the desired logic is to display the message in an event of API error, it is clearer if the method itself simply returned `true`. Ideally we will eventually want this to throw an error that can be caught and handled appropriately.